fix: add lastValidHeading state fields to AppState (mainline broken)

PR #162 introduced \`state.lastValidHeadingDeg\` / \`state.lastValidHeadingMs\` references in \`src/location.ts\` but the corresponding \`AppState\` fields were never added to \`src/types.ts\`. \`npm run type-check\` on mainline now fails with 4 TS2339 errors.

\`\`\`
src/location.ts(136,17): error TS2339: Property 'lastValidHeadingDeg' does not exist on type 'AppState'.
src/location.ts(137,17): error TS2339: Property 'lastValidHeadingMs' does not exist on type 'AppState'.
src/location.ts(141,17): error TS2339: Property 'lastValidHeadingDeg' does not exist on type 'AppState'.
src/location.ts(142,37): error TS2339: Property 'lastValidHeadingMs' does not exist on type 'AppState'.
\`\`\`

## Fix

Add to \`AppState\`:
\`\`\`ts
lastValidHeadingDeg: number | null;
lastValidHeadingMs: number;
\`\`\`

Add to \`createInitialState()\`:
\`\`\`ts
lastValidHeadingDeg: null,
lastValidHeadingMs: 0,
\`\`\`

## Acceptance

- [ ] \`npm run type-check\` clean
- [ ] \`npm run lint\` clean
- [ ] \`npm test\` clean

--- a/src/types.ts
+++ b/src/types.ts
@@ -54,6 +54,10 @@ export interface AppState {
   lastAltM: number | undefined;
   lastGpsAccuracy: number | null; // most recent GPS fix accuracy (metres)
 
+  // Heading-cone wedge: hold last valid GPS bearing for ~10s when course is NaN at low speeds
+  lastValidHeadingDeg: number | null;
+  lastValidHeadingMs: number;
+
   // Hysteresis state for GPS weak-signal badge — prevents flicker in marginal signal
   gpsWeakStreak: number;        // consecutive fixes with accuracy > TRAIL_MAX_ACCURACY_M
   gpsStrongStreak: number;      // consecutive fixes with accuracy < (TRAIL_MAX_ACCURACY_M - 5)
@@ -92,6 +96,8 @@ export function createInitialState(): AppState {
     lastSpeedMs: 0,
     lastAltM: undefined,
     lastGpsAccuracy: null,
+    lastValidHeadingDeg: null,
+    lastValidHeadingMs: 0,
     gpsWeakStreak: 0,
     gpsStrongStreak: 0,
     gpsWeakBadgeVisible: false,

